### PR TITLE
Fix typo in stake-pool help text and clarify drep queries arguments

### PR DESCRIPTION
**Pull request by carbolymer**
# Changelog

```yaml
- description: |
    Fix typo in stake-pool help text and clarify drep queries arguments
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
   - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Additional context for the PR goes here.

If the PR fixes a particular issue please provide a
[link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=)
to the issue.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] The changelog section in the PR is updated to describe the change
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Governance/Query.hs [hunk 1]
```diff
@@ -65,7 +65,7 @@ pGovernanceQueryDRepStateCmd env era = do
   pure
     $ subParser "drep-state"
     $ Opt.info (GovernanceQueryDRepStateCmd cOn <$> pDRepStateQueryCmd)
-    $ Opt.progDesc "Get the DRep state"
+    $ Opt.progDesc "Get the DRep state. If no DRep credentials are provided, return stake distributions for all of them."
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/EraBased/Options/Governance/Query.hs]: I'm not sure what the accepted terminology is here. Should it be "voting stake distributions"?

File: cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_query.cli [hunk 2]
```diff
@@ -14,6 +14,9 @@ Available options:
 Available commands:
   constitution             Get the constitution
   gov-state                Get the governance state
-  drep-state               Get the DRep state
-  drep-stake-distribution  Get the DRep stake distribution
+  drep-state               Get the DRep state. If no DRep credentials are
+                           provided, return stake distributions for all of them.
+  drep-stake-distribution  Get the DRep stake distribution. If no DRep
```
Comment (Jimbo4350) [inline cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_query.cli:19]: Likewise here. Ask @CarlosLopezDeLara for some feedback, not sure if we care to distinguish between reward and voting stake here as well. If you decide to change the parser, be sure to maintain backwards compatibility.

**Review by Jimbo4350 — APPROVED** [review]
LGTM, just a couple comments

File: cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_query.cli (hunk 2 above)
Comment (CarlosLopezDeLara) [reply cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_query.cli:19]: Ledger uses the term "DRep stake distribution" so I think we are good; introducing a different term like _voting stake distribution,_ while been more precise, could introduce noise.

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Governance/Query.hs (hunk 1 above)
Comment (carbolymer) [reply cardano-cli/src/Cardano/CLI/EraBased/Options/Governance/Query.hs]: I've used ledger terminology here, which seems to be fine according to  https://github.com/input-output-hk/cardano-cli/pull/281#discussion_r1326052618

**Pull request merged**